### PR TITLE
Ini bugfix

### PR DESCRIPTION
**Pull request by tom-wa**

### CODE REVIEW

**Comment by markus2330** [comment]
Please write (at least short) description. Make sure the text `fix #nr` is in the respective commits.

@fberlakovich: Can you also take a look at this.

File: src/plugins/ini/ini.c [hunk 1]
```diff
@@ -226,15 +228,15 @@ int elektraIniGet(Plugin *handle, KeySet *returned, Key *parentKey)
 	{
 		switch (ret)
 		{
-		case -1:
```
Comment (markus2330) [inline src/plugins/ini/ini.c]: please do not indent.

File: src/plugins/ini/ini.c [hunk 2]
```diff
@@ -84,6 +84,8 @@ static int iniKeyToElektraKey (void *vhandle, const char *section, const char *n
 
 	keyAddBaseName (appendKey, name);
 
+	if(value == NULL)
+		keySetMeta(appendKey, "ini/empty", "");
```
Comment (markus2330) [inline src/plugins/ini/ini.c]: Can a null key be used unambiguously? If yes, it should be preferred. There are already 3 states of emptyness (key not present, value not present, string emtpy), we should not over-complicate everything.

File: src/plugins/ini/inih-r29/inih.c [hunk 3]
```diff
@@ -147,8 +147,11 @@ int ini_parse_file(FILE* file,const struct IniConfig* config, void* user)
                     error = lineno;
             }
             else if (!error) {
-                /* No '=' or ':' found on name[=:]value line */
-                error = lineno;
+        	end = find_char_or_comment(start, '\0');
```
Comment (markus2330) [inline src/plugins/ini/inih-r29/inih.c]: has spaces

File: src/plugins/ini/testmod_ini.c [hunk 4]
```diff
@@ -103,7 +103,49 @@ static void test_plainIniWrite(char *fileName)
...
+			keyNew ("user/tests/ini-write/section2",
+					KEY_BINARY,
+					KEY_END),
+			keyNew ("user/tests/ini-write/section2/key3",
+					KEY_VALUE, "value3",
+					KEY_END),
+			keyNew ("user/tests/ini-write/section2/emptykey", KEY_META, "ini/empty", "", KEY_END),
+			KS_END);
+
+	succeed_if(plugin->kdbSet (plugin, ks, parentKey) >= 1,
```
Comment (markus2330) [inline src/plugins/ini/testmod_ini.c]: could you test full round-trip? also get it again and also check if keyset is equal.

**Comment by fberlakovich** [comment]
Looks nice. I am very happy that this was finally fixed and even with a much simpler solution than I could think of.

File: src/plugins/ini/ini.c (hunk 2 above)
Comment (tom-wa) [inline src/plugins/ini/ini.c]: i couldn't really differentiate between 
`key = ` and `key` without the metakey when writing back to the config file

Comment (markus2330) [inline src/plugins/ini/ini.c]: tom-wa wrote:
> i couldn't really differentiate between 
> `key = ` and `key` without the metakey when writing back to the config file

Ok, I see. So is the following correct?

- null keys are section (status quo)
- `key` is an empty string with ini/empty set
- `key = ` is an empty string

Please correct my statements and add them to README.md.
You also need to rebase, I modified the README.md.

Comment (tom-wa) [inline src/plugins/ini/ini.c]: Yes, you are correct. 
Will rebase and update.

**Pull request merged**